### DISCUSSION TITLE
How to allow page/route only be loaded in iframe?

### DISCUSSION THREAD

**Discussion by andorfermichael** [nuxt/nuxt#9364] [Questions (Nuxt 2)] [answered]
For example, I have an index page reachable via https://example.com which shows a chatbot and a footer component. Furthermore, I have a chatbot page that only contains the chatbot because it will be embedded as an iframe and I don't want the footer etc. to be loaded.

However, I don't want users of my web app to be able to visit https://example.com/chatbot.

My question is if there is an option to allow a page/route to be only loaded in an iframe?

Do you know of another approach for integrating my chatbot component in an iframe without the need of a page?

**Comment by TheAlexLichter** [comment, marked as answer]
You can check on client-side if your page is loaded in an iframe by comparing `window.top` to `window.self`. This is only a loose protection and can be avoided easily though.

Example fn:

```js
function isInIFrame () {
  try {
    return window.self !== window.top;
  } catch (e) {
    return true; // Error when reading window.top in iFrame
  }
}
```
